plugins/org.locationtech.udig.render.feature.basic/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureRenderer.java

```diff
@@ -161,7 +161,7 @@ private void prepareDraw( IProgressMonitor monitor ) throws IOException, SchemaE
         SimpleFeatureType schema = featureSource.getSchema();
 
         // Original Query provided by Layer.getFilter() as adjusted by selection and edit filter
-        Query query = getContext().getFeatureQuery();
+        Query query = new Query(getContext().getFeatureQuery());
```

Comment:
in case Query.ALL is returned its not allowed to modify it afterwards. So the fix is tro create a new one with the Filter and append additional criterias afterwards.